src/EditorFeatures/Text/Implementation/TextBufferFactoryService/TextBufferCloneServiceFactory.cs

```diff
@@ -16,25 +16,27 @@ internal class TextBufferCloneServiceFactory : IWorkspaceServiceFactory
         [ImportingConstructor]
         public TextBufferCloneServiceFactory(
             ITextBufferFactoryService textBufferFactoryService,
-            IContentTypeRegistryService contentTypeRegistry)
+            IContentTypeRegistryService contentTypeRegistryService)
         {
-            _singleton = new TextBufferCloneService((ITextBufferFactoryService3)textBufferFactoryService, contentTypeRegistry.UnknownContentType);
+            _singleton = new TextBufferCloneService((ITextBufferFactoryService3)textBufferFactoryService, contentTypeRegistryService);
         }
 
         public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
         {
             return _singleton;
```

Comment:
_singleton seems a misnomer here: there's one you'll get through the factory, and one directly from the export below. This could just have the [ImportingConstructor] import the type below and return it, just making this an even thinner shim.

(If the only thing still using the workspace service is that inline rename thing, then just disregard this comment and we'll delete this then.)

Reply:
➡️ Covered as part of #25659